wallets/server.js

```diff
@@ -6,32 +6,55 @@ import * as lnbits from '@/wallets/lnbits/server'
 import * as nwc from '@/wallets/nwc/server'
 import * as phoenixd from '@/wallets/phoenixd/server'
 import * as blink from '@/wallets/blink/server'
+import * as bolt12 from '@/wallets/bolt12/server'
 
 // we import only the metadata of client side wallets
 import * as lnc from '@/wallets/lnc'
 import * as webln from '@/wallets/webln'
 
 import { walletLogger } from '@/api/resolvers/wallet'
 import walletDefs from '@/wallets/server'
-import { parsePaymentRequest } from 'ln-service'
+import { parseInvoice } from '@/lib/boltInvoices'
+import { isBolt12Offer, isBolt12Invoice } from '@/lib/bolt12'
 import { toPositiveBigInt, toPositiveNumber, formatMsats, formatSats, msatsToSats } from '@/lib/format'
 import { PAID_ACTION_TERMINAL_STATES, WALLET_CREATE_INVOICE_TIMEOUT_MS } from '@/lib/constants'
 import { timeoutSignal, withTimeout } from '@/lib/time'
 import { canReceive } from './common'
 import wrapInvoice from './wrap'
 
-export default [lnd, cln, lnAddr, lnbits, nwc, phoenixd, blink, lnc, webln]
+export default [lnd, cln, lnAddr, lnbits, nwc, phoenixd, blink, lnc, webln, bolt12]
 
 const MAX_PENDING_INVOICES_PER_WALLET = 25
 
-export async function createInvoice (userId, { msats, description, descriptionHash, expiry = 360 }, { predecessorId, models }) {
+async function checkInvoice (invoice, { msats }, { lnd, logger }) {
+  const parsedInvoice = await parseInvoice({ lnd, request: invoice })
+  logger.info(`created invoice for ${formatSats(msatsToSats(parsedInvoice.mtokens))}`, {
+    bolt11: invoice
+  })
```

Comment:
I think this logger call should be in `createInvoice` immediately after `walletCreateInvoice` like it was before